ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/TestWorker.java

```diff
@@ -1026,6 +1029,38 @@ public void testFindNextCompactThrowsTException() throws Exception {
     verify(msc, times(0)).markFailed(any());
   }
 
+  @Test
+  public void testDoesntGatherStatsIfCompactionFails() throws Exception {
+    try (
```

Comment:
wrap with `try` only what's really needed, not the whole test setup

Reply:
Refactored.